leptonai/cli/job.py

```diff
@@ -201,7 +202,9 @@ def create(
     elif not job_spec.container.image:
         job_spec.container.image = BASE_IMAGE
     if port:
-        job_spec.container.ports = port
+        job_spec.container.ports = [
+            ContainerPort.make_container_port_from_string(p) for p in port
```

Comment:
Main fix is this line: instead of the old behavior (storing the raw port string), we actually make the ContainerPort struct.